src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,146 @@
+                PreparedStatement preparedStatement = connection
+                           .prepareStatement(SQL_CREATE_REQUEST, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(FIRST_PARAMETER_INDEX, book.getTitle());
+            preparedStatement.setBigDecimal(SECOND_PARAMETER_INDEX, book.getPrice());
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to "
+                        + "inserted at least one row, but inserted 0 rows.");
+            }
+
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_COLUMN_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish connection"
+                    + " with the database", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Optional<Book> optionalBook = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                           .prepareStatement(SQL_BY_ID_REQUEST)) {
+            preparedStatement.setLong(FIRST_PARAMETER_INDEX, id);
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (resultSet.next()) {
+                Long identify = resultSet.getObject(FIRST_COLUMN_INDEX, Long.class);
+                String title = resultSet.getObject(SECOND_COLUMN_INDEX, String.class);
+                BigDecimal price = resultSet.getObject(THIRD_COLUMN_INDEX, BigDecimal.class);
+
+                Book book = new Book(identify, title, price);
+                optionalBook = Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish connection"
+                    + " with the database", e);
+        }
+        return optionalBook;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SQL_OUTPUT_REQUEST)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(FIRST_COLUMN_INDEX, Long.class);
+                String title = resultSet.getObject(SECOND_COLUMN_INDEX, String.class);
+                BigDecimal price = resultSet.getObject(THIRD_COLUMN_INDEX, BigDecimal.class);
+
+                Book book = new Book(id, title, price);
+                bookList.add(book);
```

Comment:
Again, to avoid code duplication, you should create a separate method to convert `ResultSet` to `Book` object and reuse it in both `findById` and `findAll` methods.